examples/run/linenoise.h

```diff
@@ -0,0 +1,113 @@
+/* linenoise.h -- VERSION 1.0
+ *
+ * Guerrilla line editing library against the idea that a line editing lib
+ * needs to be 20,000 lines of C code.
+ *
+ * See linenoise.c for more information.
+ *
+ * ------------------------------------------------------------------------
+ *
+ * Copyright (c) 2010-2023, Salvatore Sanfilippo <antirez at gmail dot com>
+ * Copyright (c) 2010-2013, Pieter Noordhuis <pcnoordhuis at gmail dot com>
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *  *  Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *
+ *  *  Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
```

Comment:
Note that this will require adding a notice in the releases.

Reply:
Not sure how this is usually handled, I guess we could add a file with all 3rd party copyright notices and copy it to the release packages.

Reply:
We have the same issue with CURL if we start distributing it for Windows FWIW, from curl license: "Permission to use, copy, modify, and distribute this software for any purpose with or without fee is hereby granted, provided that the above copyright notice and this permission notice appear in all copies."

@ggerganov I you were speaking about shipping curl on Windows recently.

Reply:
I also don't know what is the common way to handle this. Probably look for how other projects do it.

Btw, each `llama.cpp` release already includes a copy of the full source code in this repo so I think all licenses are technically already included in the release packages? Again, we can make this more explicit if necessary - feel free to improve.

Reply:
I'm ok with that technique, there's no standard technique. Everybody does it a little different.

Reply:
I am not a lawyer, but I am not convinced that's enough. We already copy our own `LICENSE` file to the release packages as `llama.cpp.txt`, so it shouldn't be a problem to also add the 3rd party licenses there.

Reply:
@slaren can you show me the script where that copy occurs? We can just check in:

https://github.com/ericcurtin/linenoise.cpp/blob/main/LICENSE

and copy that file to linenoise.cpp.txt maybe ?

Reply:
https://github.com/ggerganov/llama.cpp/blob/master/.github/workflows/build.yml

Reply:
Thanks my searching tool skips hidden directories, sometimes I forget this. I hope it's good now, it should do the copy of the LICENSE now I think.